kyaml/kio/kio.go

```diff
@@ -10,6 +10,8 @@ import (
 	"sigs.k8s.io/kustomize/kyaml/yaml"
 )
 
+const NoProcessAnnotation = "config.kubernetes.io/no-process"
```

Comment:
Is there a place we list all of the annotations?  If not, would you create a separate PR that documents each of the annotations and what they do.

WDYT of refactoring all of the annotations into a single package that is well documented with godoc -- e.g. `sigs.k8s.io/kustomize/kyaml/annotations`.  There are only about 1/2 dozen right now, so seems like the right time to do it.

Reply:
@seans3 @pwittrock This seems to have a wider blast radius across all areas of kpt such as pkg, cfg(including auto-setters) and functions. If this is for just to not process inventory object by functions, then probably add a special case for that. Currently this seems like No-process for all commands if annotation is set, which I am not sure we are ready yet and should offer it without broader discussion and possible implications.

Reply:
How does this mesh with`config.kubernetes.io/local-config` annotation? It uses in a lot of examples (especially on function configs).

+1 to centralizing and documenting different annotations and their semantics.